server/src/main/java/org/elasticsearch/index/mapper/FieldTypeLookup.java

```diff
@@ -68,21 +68,29 @@
         }
         this.maxParentPathDots = maxParentPathDots;
 
+        Map<String, MappedFieldType> resolvedRuntimeMappers = new HashMap<>();
+        Map<String, DynamicFieldType> resolvedDynamicMappers = new HashMap<>();
         for (FieldAliasMapper fieldAliasMapper : fieldAliasMappers) {
             String aliasName = fieldAliasMapper.name();
             String path = fieldAliasMapper.path();
-            MappedFieldType fieldType = fullNameToFieldType.get(path);
-            fullNameToFieldType.put(aliasName, fieldType);
-            if (fieldType instanceof DynamicFieldType) {
-                dynamicFieldTypes.put(aliasName, (DynamicFieldType) fieldType);
+            RuntimeField aliasField = new AliasRuntimeField(aliasName, path);
+            MappedFieldType resolved = aliasField.asMappedFieldType(fullNameToFieldType::get);
```

Comment:
This means that the internal representation of an alias field as it existed so far is the same as the runtime alias field, but this has no effect on how it gets serialized and sent out throughout the nodes and stored in the cluster state, correct?